mobify/platform-scaffold
# Fix case where newsletter loading spinner never goes away

Use more nuanced Push Messaging selectors to determine loading spinner behavior

 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/884

## Changes
- Removes use of `isMessagingReady` selector and replaces with use of `getStatus` selector
- Uses new `isSupported` selector instead of the utility method
- Swaps around the ternary render so that if messaging is *not* supported, or the status is ever `FAILED`, we simply render the newsletter form

## Todos

## How to test-drive this PR
- Check out and run `npm link && npm run prod:build` from https://github.com/mobify/progressive-web-sdk/pull/884
- `npm link progressive-web-sdk`
- Inside `web/package.json`, change the `messagingSiteId` value to gibberish
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- You should not see the loading spinner indefinitely - it should change to just the newsletter form (i.e. no tabs)
